src/features/lightspeed/inlineSuggestions.ts

```diff
@@ -33,6 +33,7 @@ import { getAdditionalContext } from "./inlineSuggestion/additionalContext";
 let inlineSuggestionData: InlineSuggestionEvent = {};
 let inlineSuggestionDisplayTime: Date;
 let previousTriggerPosition: vscode.Position;
+let currentSuggestion: string | undefined;
```

Comment:
It would feel more natural to me to store the whole answer (`CompletionResponseParams` type), instead of just one single field. Also we can above the `undefined` and just initalize the value with an empty `CompletionResponseParams`.
Even if we use one single attribute, this would reduce the number of objects in use at the same time.

The fact we rely on these global variables is a bit annoying, but I don't see an easy solution to avoid that.

Reply:
@goneri  Thanks for your comments. I have looked at the code more closely and now think that we'd better make `insertTexts` global instead of `currentSuggestion`.  In the initial code, I found I needed to run `trim()` before comparing strings, but that was because we call `adjustInlineSuggestionIndent()` before setting `insertTexts`.  By using `insertTexts` we can get rid of `trim()`, which I think would generate incorrect results depending on indents in the code. I will update the code in that way.

> The fact we rely on these global variables is a bit annoying, but I don't see an easy solution to avoid that.

I do not like dependence on global variables either. I am feeling these coding is more like application development on MS-DOS :-)